src/components/voting/VoterButtons.tsx

```diff
@@ -1,68 +1,82 @@
 import { BareProps } from '../utils/types'
 import { IconWithLabel } from '../utils'
 import { useResponsiveSize } from '../responsive'
-import { idToPostId, PostStruct } from 'src/types'
-import { useSubscribedPost } from '../posts/view-post'
+import { PostStruct } from 'src/types'
 import { useGetReloadPost } from 'src/rtk/app/hooks'
+import { Reaction, ReactionEnum, ReactionId, ReactionStruct, ReactionType, selectPostReactionByPostId } from 'src/rtk/features/reactions/postReactionsSlice'
+import { ButtonProps } from 'antd/lib/button'
+import { useGetUpsertReaction } from 'src/rtk/features/reactions/postReactionsHooks'
+import { useAppSelector } from 'src/rtk/app/store'
+import { getNewIdsFromEvent } from '../substrate'
 
 const TxButton = dynamic(() => import('../utils/TxButton'), { ssr: false })
 
-const log = newLogger('VoterButtons')
-
 type VoterProps = BareProps & {
   post: PostStruct,
   preview?: boolean
 }
 
-type ReactionType = 'Upvote' | 'Downvote'
-
-type VoterButtonProps = VoterProps & {
-  reactionType: ReactionType,
+type VoterButtonProps = VoterProps & ButtonProps & {
+  reactionEnum: ReactionEnum,
   reaction?: Reaction,
   onSuccess?: () => void,
   preview?: boolean
 };
 
 const VoterButton = ({
-  reactionType,
+  reactionEnum,
   reaction,
   post: { id, upvotesCount, downvotesCount },
   className,
   style,
   onSuccess,
-  preview
+  preview,
+  disabled
 }: VoterButtonProps) => {
 
   const { isMobile } = useResponsiveSize()
-  const kind = reaction ? reaction && reaction.kind.toString() : 'None'
-  const isUpvote = reactionType === 'Upvote'
+  const upsertReaction = useGetUpsertReaction()
+  const { reactionId, kind = 'None' } = reaction || { id } as ReactionStruct
+  const reactionType = reactionEnum.valueOf() as ReactionType
+  const isUpvote = reactionType === ReactionEnum.Upvote
   const count = isUpvote ? upvotesCount : downvotesCount
 
   const buildTxParams = () => {
     if (reaction === undefined) {
       return [ id, new ReactionKind(reactionType) ]
     } else if (kind !== reactionType) {
-      return [ id, reaction.id, new ReactionKind(reactionType) ]
+      return [ id, reactionId, new ReactionKind(reactionType) ]
     } else {
-      return [ id, reaction.id ]
+      return [ id, reactionId ]
     }
   }
 
   const isActive = kind === reactionType
+
+  if (id === '482') {
+    console.log('kind', kind, kind === reactionType, isActive)
+  }
+
   const color = isUpvote ? '#00a500' : '#ff0000'
 
-  const changeReactionTx = kind !== reactionType
+  const isUpdate = kind !== reactionType
+  const changeReactionTx = isUpdate
     ? 'reactions.updatePostReaction'
     : 'reactions.deletePostReaction'
 
+  const updateOrDelete = useCallback((update: boolean, _reactinoId?: ReactionId) => {
+    const newReactionId = _reactinoId || reactionId
+    const newReaction: Reaction | undefined = update
+      ? { reactionId: newReactionId || `fakeId-${id}`, kind: reactionType }
```

Comment:
I think it's much better to add a current timestamp:
```suggestion
      ? { reactionId: newReactionId || `fakeId-${id}-${new Date().getTime()}`, kind: reactionType }
```